### PR TITLE
feat: Token details page hints

### LINKED ISSUE #48: Token details page hints
### Please describe the problem that your request should resolve.
Implement hints for the token details page including a warning about the value and market cap being only theoretical. Due to how the price is calculated (comparison of two reserves), somebody can artificially inflate the price of a worthless token.
Events data can be explained as generic.

### Describe the solution you'd like
Probably hint messages showing in popup messages after clicking hint icons

### Is it already possible to achieve the same outcome in any other way? If so, how?
No

### PR DESCRIPTION
**Pull request by janmichek**
## Description
resolves #48

## Checklist:

- [x] I have read and followed the [Contributing Guide](../../CONTRIBUTING.md)  
- [x] add hints to token details

### CODE REVIEW

File: src/utils/hints/tokensHints.js [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
   tokenSymbol: 'Non-unique symbol of the token.',
   tokenName: 'Full name of the token.',
-  tokenContractId: 'Identifier of a the Smart Contract that implements AEX-9 interface.',
+  tokenContractId: 'Identifier of the Smart Contract that implements AEX-9 interface.',
+  price: 'Fiat price of one token.',
+  marketCap: 'Total fiat market value of all tokens.',
+  totalSupply: 'Amount of tokens minted.',
+  decimals: 'The decimal granularity of each token.',
+  extensions: 'Properties of the tokens.',
+  apiLinks: 'Properties of the tokens.',
```
Comment (lukeromanowicz) [inline src/utils/hints/tokensHints.js:10]: How about:
"Links to sources of the token data."

File: src/utils/hints/tokensHints.js [hunk 2]
```diff
@@ -1,5 +1,17 @@
...
+  account: 'Account of the token holder.',
+  amount: 'Amount of the tokens held by its owner.',
+  percentage: 'Percentage of the token held by its owner.',
```
Comment (lukeromanowicz) [inline src/utils/hints/tokensHints.js:13]: I'd add "of the token *supply* held by *an account*"

File: src/utils/hints/tokensHints.js [hunk 3]
```diff
@@ -1,5 +1,17 @@
...
+  hash: 'The hash of the transaction produced by sending the token.',
```
Comment (lukeromanowicz) [inline src/utils/hints/tokensHints.js:14]: How about replacing "produced" with "created", "dispatched", "committed"? "producing" in context of transaction creation sounds a little exotic to me but that's subjective. Feel free to ignore it

**Review by lukeromanowicz — CHANGES_REQUESTED** [review]
Really nice, I only have a few minor suggestions. Also the demo is missing =)

File: src/utils/hints/tokensHints.js [hunk 4]
```diff
@@ -1,5 +1,17 @@
 export const tokenHints = {
   tokenSymbol: 'Non-unique symbol of the token.',
   tokenName: 'Full name of the token.',
-  tokenContractId: 'Identifier of a the Smart Contract that implements AEX-9 interface.',
+  tokenContractId: 'Identifier of the Smart Contract that implements AEX-9 interface.',
+  price: 'Fiat price of one token.',
```
Comment (michele-franchi) [inline src/utils/hints/tokensHints.js:5]: To not make confusion I would suggest: 
`Fiat price of the token.`

File: src/utils/hints/tokensHints.js [hunk 5]
```diff
@@ -1,5 +1,17 @@
 export const tokenHints = {
   tokenSymbol: 'Non-unique symbol of the token.',
   tokenName: 'Full name of the token.',
-  tokenContractId: 'Identifier of a the Smart Contract that implements AEX-9 interface.',
+  tokenContractId: 'Identifier of the Smart Contract that implements AEX-9 interface.',
```
Comment (michele-franchi) [inline src/utils/hints/tokensHints.js:4]: ...`implements the AEX-9 interface.`

File: src/utils/hints/tokensHints.js [hunk 6]
```diff
@@ -1,5 +1,17 @@
...
+  price: 'Fiat price of one token.',
+  marketCap: 'Total fiat market value of all tokens.',
```
Comment (michele-franchi) [inline src/utils/hints/tokensHints.js:6]: I would suggest:
`Total market value of a token calculated by multiplying the current price by the total circulating supply.`

File: src/utils/hints/tokensHints.js (hunk 3 above)
Comment (michele-franchi) [reply src/utils/hints/tokensHints.js:14]: It's not always a "send" event so I would suggest to add a generic description, for example:
"Transaction hash of the event".

File: src/utils/hints/tokensHints.js [hunk 7]
```diff
@@ -1,5 +1,17 @@
...
+  totalSupply: 'Amount of tokens minted.',
+  decimals: 'The decimal granularity of each token.',
+  extensions: 'Properties of the tokens.',
+  apiLinks: 'Properties of the tokens.',
+  account: 'Account of the token holder.',
+  amount: 'Amount of the tokens held by its owner.',
+  percentage: 'Percentage of the token held by its owner.',
+  hash: 'The hash of the transaction produced by sending the token.',
+  eventName: 'Name of an event produced with this token.',
```
Comment (michele-franchi) [inline src/utils/hints/tokensHints.js:15]: I would simply change it "Name of the event" and also describe N/A values (not part of the standard AEX-9 interface).

File: src/utils/hints/tokensHints.js [hunk 8]
```diff
@@ -1,5 +1,17 @@
...
+  date: 'Estimated date, time, and block height of the transaction with the token.',
```
Comment (michele-franchi) [inline src/utils/hints/tokensHints.js:16]: I would remove "with the token", it sounds confusing to me.

**Review by michele-franchi — CHANGES_REQUESTED** [review]
Please have a look at my suggestions.

**Comment by janmichek** [comment]
@michele-franchi @lukeromanowicz all suggestions accepted. Thanks for your collaboration.

**Review by lukeromanowicz — APPROVED** [review]
Thank you, looks really good :+1:

File: src/utils/hints/tokensHints.js [hunk 9]
```diff
@@ -1,5 +1,17 @@
 export const tokenHints = {
   tokenSymbol: 'Non-unique symbol of the token.',
   tokenName: 'Full name of the token.',
-  tokenContractId: 'Identifier of a the Smart Contract that implements AEX-9 interface.',
+  tokenContractId: 'Identifier of the Smart Contract that implements the AEX-9 interface.',
+  price: 'Fiat price of the token.',
+  marketCap: 'Total market value of the token calculated by multiplying the current price by the total circulating supply.',
+  totalSupply: 'Amount of tokens minted.',
+  decimals: 'The decimal granularity of each token.',
+  extensions: 'Properties of the tokens.',
```
Comment (michele-franchi) [inline src/utils/hints/tokensHints.js:9]: `Properties of the token.`

File: src/utils/hints/tokensHints.js [hunk 10]
```diff
@@ -1,5 +1,17 @@
...
+  apiLinks: 'Links to sources of the token data.',
+  account: 'Account of the token holder.',
+  amount: 'Amount of the tokens held by its owner.',
+  percentage: 'Percentage of the token supply held by the account.',
+  hash: 'Transaction hash of the event.',
+  eventName: 'Name of the event. The "N/A" value means that it\'s not a part of the standard AEX-9 interface.',
+  date: 'Estimated date, time, and block height of the transaction.',
```
Comment (michele-franchi) [inline src/utils/hints/tokensHints.js:16]: `Block height and estimated date and time of the transaction.`

**Review by michele-franchi — APPROVED** [review]
Looks good, I added two new suggestions.

**Comment by CharisSiarampalis** [comment]
There is a missing hint for the Smart Contract ID
image
Also some hint are missing the triangle underneath them of its not alligned.
image

**Comment by michele-franchi** [comment]
@janmichek seems also to be broken on mobile, different hints are empty:
image

File: src/utils/hints/tokensHints.js (hunk 9 above)
Comment (janmichek) [reply src/utils/hints/tokensHints.js]: fixed

**Comment by janmichek** [comment]
> @janmichek seems also to be broken on mobile, different hints are empty: image

fixed

**Review by CharisSiarampalis — APPROVED** [review]

**Pull request merged**